public/js/utils/source-documents.js

```diff
@@ -0,0 +1,24 @@
+let sourceDocs = {};
```

Comment:
keeping state in a util is a pattern consistent with source maps, but in that case there is state in a worker. I was reluctant to subscribe to the store or move the state into the reducer as it is not serializable. But thoughts are welcome